src/genpy/dynamic.py

```diff
@@ -190,6 +190,7 @@ def generate_dynamic(core_type, msg_cat):
             messages[t] = getattr(mod, _gen_dyn_name(pkg, s_type))
         except AttributeError:
             raise MsgGenerationException("cannot retrieve message class for %s/%s: %s"%(pkg, s_type, _gen_dyn_name(pkg, s_type)))
+        messages[t]._spec = specs[t]
```

Comment:
The instance is a generated class. It doesn't have that attribute before. `_full_text` contains the plain text of the message, `_spec` contains the parsed [MsgSpec](https://github.com/ros/genmsg/blob/490b3753213ed3e2251acd218c62224b2787ee0e/src/genmsg/msgs.py#L228) instance for the message.